### ISSUE TITLE
IIIF Viewer Stuck with Page Turning at Initial Launch

### ISSUE THREAD

**Issue by mdalmau** [IUBLibTech/newton_chymistry#82]
I was trying to switch between pages in ALCH00042, and I couldn’t get it to switch to another page by any means other than exiting the viewing window and scrolling down to the next place where I could click for an image.  Reported by Alex Wingate.

**Comment by mdalmau** [comment]
I tested this: http://possum.dlib.indiana.edu:8197/text/ALCH00042/ and I got the same error.  I clicked on the first page image icon, and was unable to get to the second page (either using the arrows or the thumbnails).  I got the IIIF spinning cube. If I click on the 2nd page icon, the viewer will work.  There's seems to be an initial loading problem.  @Conal-Tuohy, any ideas?

**Comment by Conal-Tuohy** [comment]
I've made an attempt to fix this bug; if you deploy the b89c3370e0f2088dc009890fc9f396a77de3188a revision I think it should all work. I was able to replicate the problem on my own VM and this fixed it for me.

**Comment by mdalmau** [comment]
@Conal-Tuohy, sounds like this will require a merge on Carbon, yes? Let me know when the pull request is up and I can follow-up with @randalldfloyd

**Comment by Conal-Tuohy** [comment]
@mdalmau I already pushed the fix to the master branch of this repo;  b89c337

**Comment by mdalmau** [comment]
Hi @Conal-Tuohy. @randalldfloyd updated the servers so I can test your fix and we both continue to encounter that initial hang-up when we first launch the IIIF viewer.  It'll open on the first image but when you clock to the 2nd image in the viewer, it hangs.  Randall checked to make sure that the JS is not cached.  We are getting the same behaviors on [carbon](http://carbon.dlib.indiana.edu:8220/text/ALCH00042/) and [possum](http://possum.dlib.indiana.edu:8197/text/ALCH00042/), which is our staging for production. I am going to update the index just in case something in the TEI changed, but I wouldn't think so.  Any ideas? Thanks!

**Comment by mdalmau** [comment]
I re-indexed.  The culprit is the "forward" or "next" arrow key.  What ever that button triggers is getting hung-up.  After I re-indexed docs on Carbon & Possum, I tried different navigation approaches.  On Carbon, after clicking on the first image link in the normalized view of ALCH00042, I used the thumbnails to navigate to the 2nd and other pages.  That worked.  On Possum, I used the "next" button and the cube continues to spin.  Not loading the 2nd page.  If you clock on other buttons like "back to very first image" or "forward to very last image" then you can navigate one page at a time, front or back, without any problems.

**Comment by mdalmau** [comment]
@Conal-Tuohy:  to help @randalldfloyd  and me with troubleshooting, I want to confirm where the manifest file is coming from?  The TEI header contains a PURL pointer in the <idno> to the Pages Online manifest BUT when I view the documents in the "directory of texts" in the admin section (this pages lists the P4, P5, HTML, Xubmit metadata, etc.) the IIIF manifest source renders as http://possum.dlib.indiana.edu:8197/iiif/ALCH00001/manifest.  It looks like the manifest file may be local to the application in this view.

**Comment by Conal-Tuohy** [comment]
The IIIF manifest is indeed being loaded directly from IU's IIIF server at https://purl.dlib.indiana.edu/iudl/newton/iiif/ALCH00042 which redirects to https://webapp1.dlib.indiana.edu/imagesrv//dd%2Fj5%2F2w%2Ff3%2F08-intermediate_file.jp2/info.json

The "admin" section of the Newton webapp has a list of IIIF manifests which are the outputs of a pipeline in the app which was used before the new IIIF server was wheeled in to replace it. Those manifests are not in use, currently.

I can see that my attempted fix has not done the trick (though it solved the issue on my machine). I will take another look and see if I can debug it. The issue here is that Universal Viewer, when it's loaded, needs to be told what it's current canvas should be. The page icon on the web page has a canvas identifier attached to it, and the idea is that when the icon is clicked, an event-handler notices the click, starts up UV, gets the canvas identifier from the clicked icon, and sets UV's current canvas index to that value. If something goes wrong with any of that, then UV doesn't know what the current canvas is. When you click the next and previous buttons in UV, it will not be able to tell which the is the appropriate canvas to move to.

**Comment by Conal-Tuohy** [comment]
I have made a new revision in which the canvas index is converted to an integer before being fed to UV. I hope this fixes the issue!

**Comment by randalldfloyd** [comment]
@Conal-Tuohy Thanks, this is my big to-do this week so thanks for helping me get close to the root cause.  If I follow your explanation and the code accurately, it looks like the identifiers attached to events are just numeric values, taken from their position in the document, and whose values should correspond to the positions of the canvases. It seems like that should just work, but I just wanted to point out that there are functional differences between our manifest and the ones derived in the app pipeline.  Most notably, there are no `thumbnail` sections inside of our canvases like there are yours, but that shouldn't matter. If I understand the JS code, it shouldn't be trying to resolve canvases by any URI ID's but rather just numeric positions.

**Comment by randalldfloyd** [comment]
@Conal-Tuohy @mdalmau I deployed this change and initial testing seems encouraging.  I have yet to experience the problem of the UV not knowing what canvas to set when clicking the page icons.  However, @mdalmau, is it expected right now that some of the objects in Pages Online do not have the correct number of pages?  In ALCH00001 there are multiple page links throughout the document but our manifest in Pages only has one canvas, and the underlying work only has one attached page.  That results in `canvas index out of bounds` in the UV because there are positions in the document where there should be pages but they don't exist in the manifest.

**Comment by mdalmau** [comment]
@randalldfloyd yes, there are missing page images.  I am meeting with Nick about this soon.  @aewingate may have uncovered some wonkiness with ALCH000200 but I first need to verify that Pages is NOT missing images.  Hang tight and I'll post an update in the next day or two.  Thanks to you & @Conal-Tuohy
